src/coroutine.js

```diff
@@ -1,25 +1,21 @@
+import Action from './Action'
+
 export default function (resolve, iterator, promise) {
 	new Coroutine(resolve, iterator, promise).run()
+	// taskQueue.add(new Coroutine(resolve, iterator, promise))
 	return promise
 }
 
-class Coroutine {
+class Coroutine extends Action {
 	constructor (resolve, iterator, promise) {
+		super(promise)
 		this.resolve = resolve
-		this.iterator = iterator
-		this.promise = promise
+		this.next = iterator.next.bind(iterator)
+		this.throw = iterator.throw.bind(iterator)
```

Comment:
I tend to avoid `bind` like the plague because of its performance implications.  I'd rather keep the "extra" `this` threading in order to avoid `bind`.  Maybe there should be a variant of `tryCall` that allows passing a `thisArg`.  Or maybe just modify `tryCall` to accept a `thisArg` as its 3rd param and use `.call`, since there's basically no perf penalty for `.call` these days.  Thoughts?